oasislmf/gen_test_data/generate_test_data.py

```diff
@@ -0,0 +1,725 @@
+import argparse
+from collections import OrderedDict
+import numpy as np
+import os
+import struct
+
+
+class ModelFile:
+    def __init__(self):
+        pass
+
+    def seed_rng(self):
+        if self.random_seed == 0:
+            np.random.seed(0)
+        elif self.random_seed == -1:
+            np.random.seed(1234)
+        else:
+            np.random.seed(self.random_seed)
+
+    def write_file(self):
+        with open(self.file_name, 'wb') as f:
+            if self.start_stats:
+                for k, v in self.start_stats.items():
+                    f.write(struct.pack(v['dtype'], v['value']))
+            for col in self.data:
+                for idx, dtype in enumerate(self.dtypes.values()):
+                    try:
+                        f.write(struct.pack(dtype, col[idx]))
+                    except TypeError:   # If line is not tuple
+                        f.write(struct.pack(dtype, col))
+
+    # Method for debugging file output
+    def debug_write_file(self):
```

Comment:
I think in this debug part, it would be interesting to put a header,
Also I would use format, something like
line_format = "{}"+", {}"*(len(self.dtypes)-1)
for line in self.data:
  print(line)